contracts/ligo/src/swaps/fa2_whitelisted_swap.mligo

```diff
@@ -0,0 +1,63 @@
+#import "../../fa2/fa2_interface.mligo" "FA2"
+#include "../../fa2_modules/admin/non_pausable_simple_admin.mligo"
+#include "fa2_swap.mligo"
+
+(* ======== Swaps whitelist component ======== *)
+
+(* ==== Types ==== *)
+
+type whitelist = (address, unit) big_map
+
+let addr_list_to_big_map(new_allowed: address list) : whitelist =
+  List.fold
+    (fun (m, a : (address, unit) big_map * address) -> Big_map.add a () m)
+    new_allowed
+    (Big_map.empty : (address, unit) big_map)
```

Comment:
Why not just take the `big_map` itself in the parameter?
It would save all the cost of folding over the list.

Reply:
Mm, this indeed will be cheaper, but accepting weirdo `big_map address unit` in a public interface seemed not good.

On the other hand, probably this contract will eventually be used only via typescript bindings, and I can still accept a list there and perform the necessary conversion. And in our tests accepting a `BigMap` should not be too painful.

Do you think it's worth doing like this?

Reply:
Yeah, I think it's worth using a `BigMap` and let interface implementation decide on how to present it.

Ultimately the saved cost should justify the "weird" `big_map` IMO, however we should explain in the README why that's the structure used (it might be picked up by someone who wants to interact with it directly)